src/main/java/org/apache/sling/feature/extension/apiregions/launcher/LauncherProperties.java

```diff
@@ -101,18 +105,22 @@ public static Properties getBundleIDtoFeaturesMap(Feature app) {
     }
 
     public static Properties getFeatureIDtoRegionsMap(ApiRegions regions) {
-        Map<ArtifactId, Set<String>> map = new HashMap<>();
+        Map<ArtifactId, List<String>> map = new HashMap<>();
 
         for (ApiRegion region : regions.listRegions())
         {
             for (ArtifactId featureId : region.getFeatureOrigins()) {
                 map.compute(featureId, (id, regionNames) -> {
                     if (regionNames == null) {
-                        regionNames = new HashSet<>();
+                        regionNames = new LinkedList<>();
                     }
                     regionNames.add(region.getName());
+                    int insertionIndex = regionNames.size() - 1;
                     for (ApiRegion parent = region.getParent(); parent != null; parent = parent.getParent()) {
-                        regionNames.add(parent.getName());
+                        String parentName = parent.getName();
+                        if (!regionNames.contains(parentName)) {
```

Comment:
This check is not needed - the ApiRegions api ensures that the name of a region is unique

Reply:
Hi @cziegeler this code iterates over multiple API Regions, which means that without the check the values on the `map` can have duplicates. It's easy to see that by removing the check, it will cause 2 test failures.
Previously this was not needed because the values on the map were of type `Set`...

Reply:
I didn't realize the outer loop, so yes, makes sense